code/drasil-docLang/lib/Drasil/DocumentLanguage.hs

```diff
@@ -423,11 +423,28 @@ mkUCsSec (UCsProg c) = SRS.unlikeChg (intro : mkEnumSimpleD c) []
 -- | Helper for making the Traceability Matrices and Graphs section.
 mkTraceabilitySec :: TraceabilitySec -> SystemInformation -> Section
 mkTraceabilitySec (TraceabilityProg progs) si@SI{_sys = sys} = TG.traceMGF trace
-  (map (\(TraceConfig _ pre _ _ _) -> foldlList Comma List pre) progs)
+  (map (\(TraceConfig _ pre _ _ _) -> foldlList Comma List pre) fProgs)
   (map LlC trace) (filter (not.isSpace) $ abrv sys) []
   where
-  trace = map (\(TraceConfig u _ desc rows cols) -> TM.generateTraceTableView
-    u desc rows cols si) progs
+    trace = map (\(TraceConfig u _ desc rows cols) -> TM.generateTraceTableView
+      u desc rows cols si) fProgs
+    fProgs = filter (\(TraceConfig _ _ _ rows cols) -> eitherEmtpy 
+      (header (showUIDs rows sidb) si) (header (showUIDs cols sidb) si)) progs
+    sidb = _sysinfodb si
+
+-- | Helper to get UIDs
+showUIDs :: [TM.TraceViewCat] -> ChunkDB -> [UID] -> [UID]
+showUIDs a s e = filter (`elem` (Map.keys $ s ^. traceTable)) $ concatMap (\x -> x e s) a
+
+-- | Helper to get headers of rows and columns
+header :: ([UID] -> [UID]) -> SystemInformation -> [Sentence]
+header f = TM.traceMHeader (f . nub . Map.keys . (^. refbyTable))
+
+-- | Helper to check if either list is empty instead of using isEmpty
+eitherEmtpy :: [a] -> [a] -> Bool
+eitherEmtpy _ [] = False
```

Comment:
`Emtpy` is a typo.

You can also use `null` to check if a list is empty instead of pattern matching.